Tests/AsyncQueueTests/AsyncQueueTests.swift

```diff
@@ -26,8 +26,149 @@ import XCTest
 
 final class AsyncQueueTests: XCTestCase {
 
-    func test_example() {
-        _ = AsyncQueue()
+    // MARK: XCTestCase
+
+    override func setUp() async throws {
+        try await super.setUp()
+
+        systemUnderTest = AsyncQueue()
+    }
+
+    override func tearDown() async throws {
+        try await super.tearDown()
+
+        await systemUnderTest.await { /* Drain the queue */ }
+    }
+
+    // MARK: Behavior Tests
+
+    func test_async_sendsEventsInOrder() {
```

Comment:
Sorry to be the nitpickiest guy that has ever nitpicked, but _snake case?_

Reply:
😅 yeah my test-naming style is: `test_methodBeingTested_whatIAmTesting`. That format helps me make my test cases more targeted.

Reply:
I respect that, don't think I've ever had to develop a test naming scheme but I can see it being a good idea.